### PR TITLE
Ian/painting

### PR DESCRIPTION
**Pull request by ianterrell**
Feel free to look at this or not as you think about it and work on it!

### CODE REVIEW

**Review by erikamaker — APPROVED** [review]
I use adjectives in my variable names so much-- thank you for pointing this out. I can see why this way is clearer.

**Review by erikamaker — APPROVED** [review]

**Review by erikamaker — APPROVED** [review]

**Review by erikamaker — APPROVED** [review]
I love what you've done with the place! 😄

**Review by erikamaker — APPROVED** [review]
This extraction makes it easier for me to process what I'm looking at-- thank you for that 😄

**Review by erikamaker — COMMENTED** [review]
That all makes good sense to me! I see almost all the same elements I was planning to work with but struggled to implement. I'm alarmed at how readable it is, but how lost I felt the other day.

**Review by erikamaker — APPROVED** [review]
I see. This adds a layer to the boxes we're drawing. Rather than simply drawing whether it's selected or unselected, it's now including the individual cells that are toggled either painted or not. It then draws the top and bottom half of the boxes like before. 

It's easy to understand. I think if I had considered including `cell` as an argument in `draw_cell`, I would have gotten there eventually (though probably not as elegantly).

File: lib/board.rb [hunk 1]
```diff
@@ -13,6 +13,9 @@ def width
     @grid.first.length
   end
 
+  def paint!(row, col)
+    @grid[row][col].paint!
```
Comment (erikamaker) [inline lib/board.rb:17]: I had to learn that `[row][col]` is the crossroads for the grid location (aka, the cell we're going to paint). I'm so used to thinking of grids differently (more like `[[]]` than `[][]`) that I didn't see what was happening initially. Just commenting on this for my own retention.

**Comment by erikamaker** [comment]
Thank you so much for laying this out in such a clear way. I feel like I have a stronger grasp of how the pieces relate to one another.

File: lib/board.rb (hunk 1 above)
Comment (ianterrell) [reply lib/board.rb:17]: It's a good thing to note! This is a common method for indexing into 2-dimensional arrays. I assume it's the most common.

**Comment by ianterrell** [comment]
> I see. This adds a layer to the boxes we're drawing. Rather than simply drawing whether it's selected or unselected, it's now including the individual cells that are toggled either painted or not. It then draws the top and bottom half of the boxes like before.

Yep, you've got it. It's just one idea and one approach. Drawing them as X's would also be one, or block characters, or changing the box style AND painting it, or any number of things really.

The neat part about the drawing being separate from the game logic is that we can still swap out the renderer and draw it differently. We can get a working version and then play with different styles to see what looks best. 🤷 

> It's easy to understand. I think if I had considered including cell as an argument in draw_cell, I would have gotten there eventually (though probably not as elegantly). Also, knowing where to include the Curses methods was an important hurdle I missed.

I have the advantage of having played with this library before and having picked it. 😆

**Pull request merged**